IoT-Projects/ESP8266/WiFi Switch/UBIDOTS_ESP8266_TRIAC_WIFI_SWITCH.ino

```diff
@@ -0,0 +1,261 @@
+      char result[100];
+      uint8_t i = 0;
+      for (i = 0; i < len - 3; i++) { 
+        result[i] = result_lv[i];
+      }
+      result[i] = '\0';
+      Serial.print("Label is: ");
+      Serial.println(result);
+      sprintf(variable_label, "%s", result);
+      break;
+    }
+  }
+}
+
+// cast from an array of chars to float value.
+float btof(byte * payload, unsigned int length) {
+  char * demo = (char *) malloc(sizeof(char) * 10);
+  for (int i = 0; i < length; i++) {
+    demo[i] = payload[i];
+  }
+  float value = atof(demo);
+  free(demo);
+  return value;
+}
+
+// State machine to use switch case
+void set_state(char* variable_label) {
+  variable = 0;
+  for (uint8_t i = 0; i < NUMBER_OF_VARIABLES; i++) {
+    if (strcmp(variable_label, variable_labels[i]) == 0) {
+      break;
+    }
+    variable++;
+  }
+  if (variable >= NUMBER_OF_VARIABLES) variable = ERROR_VALUE; // Not valid
+  
+}
+
+// Function with switch case to determine which variable changed and assigned the value accordingly to the code variable
+void execute_cases() {  
+  switch (variable) {
+    case 0:
+      estadoluz = value;
+      Serial.print("Luz: ");
+      Serial.println(estadoluz);
+      Serial.println();
+      break;
+    case ERROR_VALUE:
+      Serial.println("error");
+      Serial.println();
+      break;
+    default:
+      Serial.println("default");
+      Serial.println();
+  }
+
+}
+/****************************************
+
+ * Funcion principal
+
+ ****************************************/
+
+void setup() {
+
+  // put your setup code here, to run once:
+  pinMode(luz, OUTPUT);  
+  pinMode(boton, INPUT); 
+  ubiClient.ubidotsSetBroker("industrial.api.ubidots.com"); // Sets the broker properly for the business account
+  ubiClient.setDebug(true); // Pass a true or false bool value to activate debug messages
+  Serial.begin(115200);
+  ubiClient.wifiConnection(WIFINAME, WIFIPASS);
+  ubiClient.begin(callback);
+  if(!ubiClient.connected()){
+    ubiClient.reconnect();
+  }
+
+  char* deviceStatus = getUbidotsDevice(DEVICE_LABEL);
+
+  if (strcmp(deviceStatus, "404") == 0) {
+    ubiClient.add("bombillo", 0); //Insert your variable Labels and the value to be sent
+    ubiClient.ubidotsPublish(DEVICE_LABEL);
+    ubiClient.loop();
+  }
+
+  ubiClient.ubidotsSubscribe(DEVICE_LABEL,VARIABLE_LABEL1); //Insert the Device and Variable's Labels
+  Serial.println(variable_labels[1]);
+
+}
+
+void loop() {
+  // put your main code here, to run repeatedly:
+  if(!ubiClient.connected()){
+    ubiClient.reconnect();
+    ubiClient.ubidotsSubscribe(DEVICE_LABEL,VARIABLE_LABEL1); //Insert the Device and Variable's Labels
+  }
+  ubiClient.loop();    
+  // read the state of the switch into a local variable:
+  int reading = digitalRead(boton);
+  if (reading != lastButtonState) {
+    // reset the debouncing timer
+    lastDebounceTime = millis();
+  }
+
+  if ((millis() - lastDebounceTime) > debounceDelay) {
```

Comment:
refactor this, you should never use more than 2 if sentences.

If you need this amount of conditionals, please split this part in two different functions